angr/simos.py

```diff
@@ -729,6 +778,73 @@ def state_entry(self, **kwargs):
 
         return state
 
+    def state_tracer(self, input_content=None, magic_content=None, **kwargs):
```

Comment:
There seems to be some redundancy with this `state_tracer` and the one above. Can we have a base `SimOS.state_tracer` and  specialize it in `SimCGC` and `SimLinux`?

Reply:
We already have a base `SimOS.state_tracer`. But it does nothing, since `SimLinux` needs a `full_init_state` while `SimCGC` needs a `entry_state`. I asked about this once, and the answer is that we cannot replace `full_init_state` in `SimLinux` by `entry_state`. But I come up with another idea when writing this: can we replace `entry_state` in `SimCGC` by `full_init_state`? I don't know much about CGC so can you answer this for me?

Reply:
There is no distinction between entry_state and full_init_state in CGC, since CGC has no complex initialization. Similarly, for statically linked linux programs there is also no difference between entry_state and full_init_state besides the single step from LinuxLoader to the entry point.

Reply:
Yes, but I still think that we should keep it `full_init_state` for Linux, so tracer can work for dynamically linked progams. Let's make it `full_init_state` for CGC as well, is it ok?

Reply:
I like it. Go ahead.